ui/src/dashboards/utils/cellGetters.js

```diff
@@ -20,35 +20,63 @@ const getMostCommonValue = values => {
   return results.mostCommonValue
 }
 
+const numColumns = 12
+
+const getNextAvailablePosition = (dashboard, newCell) => {
+  const farthestY = dashboard.cells
+    .map(cell => cell.y)
+    .reduce((a, b) => (a > b ? a : b))
+
+  const bottomCells = dashboard.cells.filter(cell => cell.y === farthestY)
```

Comment:
I think this should be `cell => cell.y + cell.h > farthestY`
for the situation pictured here: 
![kapture 2018-04-16 at 11 37 17](https://user-images.githubusercontent.com/10937678/38828342-9aec2e50-416a-11e8-85db-225800784387.gif)